x-pack/plugins/security/common/types.ts

```diff
@@ -4,13 +4,7 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-/**
- * Type and name tuple to identify provider used to authenticate user.
- */
-export interface AuthenticationProvider {
```

Comment:
Mostly because I wanted to group interface with the relevant helper function `shouldProviderUseLoginForm` (like we do for the `AuthenticatedUser`) and having function in `types.ts` would look a bit weird.